docs/design.md

```diff
@@ -237,28 +237,34 @@ as custom commands don't return `Entity` or `EntityCommands`. This kind of usage
 ### Pattern
 
 When spawning an entity that can be interacted with, such as a button that can be pressed,
-register a [one-shot system](https://bevyengine.org/news/bevy-0-12/#one-shot-systems) to handle the interaction:
+use an observer to handle the interaction:
 
 ```rust
 fn spawn_button(mut commands: Commands) {
-    let pay_money = commands.register_one_shot_system(pay_money);
-    commands.button("Pay up!", pay_money);
+    // See the Widgets pattern for information on the `button` method
+    commands.button("Pay up!").observe(pay_money);
+}
+
+fn pay_money(_trigger: Trigger<OnPress>, /* Bevy query parameters */) {
```

Comment:
```suggestion
fn pay_money(_trigger: Trigger<OnPress>, /* Bevy system parameters */) {
```
Although this would be more clear with e.g. a dummy `mut money: ResMut<Money>` and then `money.0 += 100.0;` in the body.

Reply:
[`6d794b2` (#249)](https://github.com/TheBevyFlock/bevy_quickstart/pull/249/commits/6d794b242347929155499c766cf7846d1aceef8e)